nixos/modules/services/home-automation/openhab.nix

```diff
@@ -0,0 +1,372 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.openhab;
+
+  libDir = "/var/lib/openhab";
+
+  fileExt = type:
+    if type == "sitemaps" then "sitemap"
+    else if type == "transform" then "map"
+    else if type == "persistence" then "persist"
+    else if type == "services" then "cfg"
+    else type;
+
+  filePath = type: name:
+    "${type}/${name}.${fileExt type}";
+
+  toText = value:
+    if builtins.isList value
+    then lib.concatStringsSep "," value
+    else if builtins.isBool value
+      then if value then "true" else "false"
+      else toString value;
+
+  attrsToFile = attrs: name: (pkgs.writeText "openhab-${name}" ''
+    ${lib.concatStringsSep "\n" (lib.mapAttrsToList (key: val:
+      "${key}=${toText val}"
+    ) attrs)}
+  '');
+
+  cfgDrv = pkgs.stdenv.mkDerivation rec {
+    name = "openhab-config";
+    buildCommand = ''
+      dir=$out/etc/openhab
+
+      # conf/
+      ${lib.concatStringsSep "\n" (lib.mapAttrsToList (type: items: ''
+        ${lib.concatStringsSep "\n" (lib.mapAttrsToList (item: contents:
+          "install -Dm644 ${pkgs.writeText "${item}.${fileExt type}" contents} $dir/conf/${filePath type item}"
+        ) items)}
+      '') cfg.configFiles.conf)}
+
+      # conf/services/addons.cfg
+      install -Dm644 ${attrsToFile cfg.initialAddons "addons.cfg"} $dir/conf/services/addons.cfg
+
+      # userdata/
+      ${lib.concatStringsSep "\n" (map (e:
+        "install -Dm644 ${pkgs.writeText (builtins.baseNameOf e.name) e.contents} $dir/userdata/${e.name}"
+      ) cfg.configFiles.userData)}
+    '';
+  };
+
+in {
+  meta.maintainers = with maintainers; [ peterhoeg ];
+
+  options.services.openhab = {
+    enable = mkEnableOption "OpenHAB - home automation";
+
+    package = mkOption {
+      default = pkgs.openhab;
+      type = types.package;
+      example = literalExample ''
+        pkgs.openhab.override {
+          withAddons = true;
```

Comment:
The `openhab` package doesn't appear to have a `withAddons` attribute. Did it get removed, but this documentation hasn't been updated?

Reply:
I'm using this module with `withAddons` - it's definitely there.